processors/src/utilities.cxx

```diff
@@ -93,87 +94,82 @@ Track* utils::buildTrack(EVENT::Track* lc_track,
     if (gbl_kink_data) {
         // Instantiate an LCRelation navigator which will allow faster access 
         // to GBLKinkData object
-        UTIL::LCRelationNavigator* gbl_kink_data_nav 
-            = new UTIL::LCRelationNavigator(gbl_kink_data);
-
+        std::shared_ptr<UTIL::LCRelationNavigator> gbl_kink_data_nav = std::make_shared<UTIL::LCRelationNavigator>(gbl_kink_data);
+        
         // Get the list of GBLKinkData associated with the LCIO Track
         EVENT::LCObjectVec gbl_kink_data_list 
             = gbl_kink_data_nav->getRelatedFromObjects(lc_track);
-
+        
         // The container of GBLKinkData objects should only contain a 
         // single object. If not, throw an exception
         if (gbl_kink_data_list.size() != 1) { 
             throw std::runtime_error("[ TrackingProcessor ]: The collection " 
-                    + std::string(Collections::TRACK_DATA_REL)
-                    + " has the wrong data structure."); 
+                                     + std::string(Collections::TRACK_DATA_REL)
```

Comment:
I think we have indentation issues here :-( and elsewhere. Can you set your editor to auto indent 4 spaces?

Reply:
Hi Maurik,

I did fix my emacs config to have 4 spaces today. I haven't indented the whole file yet.Let me fix it